tests/network/vsftpd.pm

```diff
@@ -0,0 +1,50 @@
+# SUSE's openQA tests
+#
+# Copyright © 2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: vsftpd QAM regression testsuite
+#          not testing SSL now due to https://bugzilla.suse.com/show_bug.cgi?id=1116571
+# Maintainer: Jozef Pupava <jpupava@suse.com>
+
+use base 'consoletest';
+use testapi;
+use strict;
+use lockapi;
+use utils 'zypper_call';
+
+sub run {
+    select_console 'root-console';
+    if (check_var('HOSTNAME', 'server')) {
+        barrier_create('VSFTPD_FINISHED', 2);
+    }
+    # export slenkins variables
+    assert_script_run 'export SERVER=10.0.2.101';
+    assert_script_run 'export CLIENT=10.0.2.102';
+    zypper_call 'in vsftpd';
+    # copy ssh keys
+    assert_script_run 'ssh-keygen -f /root/.ssh/id_rsa -N ""';
+    type_string "ssh-copy-id -o StrictHostKeyChecking=no \$SERVER\n";
+    wait_still_screen 3;
+    type_password;
+    send_key 'ret';
+    wait_still_screen 3;
+    type_string "ssh-copy-id -o StrictHostKeyChecking=no \$CLIENT\n";
+    wait_still_screen 3;
+    type_password;
+    send_key 'ret';
+    wait_still_screen 3;
+    assert_script_run 'cd /tmp';
+    assert_script_run 'wget ' . data_url('qam/vsftpd.tar.gz');
+    assert_script_run 'tar xzfv vsftpd.tar.gz';
+    if (check_var('HOSTNAME', 'server')) {
+        assert_script_run 'bash run.sh', 300;
+    }
+    barrier_wait('VSFTPD_FINISHED');
```

Comment:
I think this barrier can be replaced with simple wait_for_children call at the end
use mmapi 'wait_for_children';
```suggestion
    wait_for_children if check_var('HOSTNAME', 'server');
```

Reply:
I will leave barrier there, for `wait_for_children` I have to first use get_children at the end it's not simpler than barrier

Reply:
You don't need get_children first (or I missed something) - anyway, keep whatever you prefer :)

Reply:
problem is that children end before server is done with testing, barrier is fine

Reply:
Seems I misunderstood, sorry - I thought client is the one doing testing and server should wait for it..